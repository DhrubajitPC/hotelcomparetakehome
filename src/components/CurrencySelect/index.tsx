import React, { FC } from "react";

const CurrencySelect: FC = () => {
  const currencies = ["USD", "SGD", "CNY", "KRW"];
  return (
    <div>
      Currency:{" "}
      <select name="currency">
        {currencies.map((c) => (
          <option value={c} key={c}>
            {c}
          </option>
        ))}
      </select>
    </div>
  );
};

export default CurrencySelect;
