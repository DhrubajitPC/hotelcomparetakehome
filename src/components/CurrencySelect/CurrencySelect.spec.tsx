import { fireEvent, render } from "@testing-library/react";
import React from "react";
import CurrencySelect from ".";

let mockFetch = jest.fn();
jest.mock("../../state/Context", () => ({
  __esModule: true,
  useAPI: () => ({
    fetchPrices: mockFetch,
  }),
  useAppContext: () => ({
    state: {
      selectedCurrency: "CNY",
    },
  }),
}));

describe("CurrencySelect", () => {
  it("should render correctly", () => {
    const { container } = render(<CurrencySelect />);
    expect(container).toMatchSnapshot();
    expect(container.getElementsByTagName("select")[0]).toHaveValue("CNY");
    expect(container.getElementsByTagName("select")[0]).not.toHaveValue("USD");
  });

  it("should request new price data when selection is changed", () => {
    const { container } = render(<CurrencySelect />);

    fireEvent.change(container.getElementsByTagName("select")[0], {
      target: { value: "SGD" },
    });
    expect(mockFetch).toHaveBeenCalledWith("SGD");
  });
});
