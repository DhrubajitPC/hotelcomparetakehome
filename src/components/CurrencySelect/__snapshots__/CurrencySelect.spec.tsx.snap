// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencySelect should render correctly 1`] = `
<div>
  <div>
    Currency:
     
    <select
      name="currency"
    >
      <option
        value="SGD"
      >
        SGD
      </option>
      <option
        value="USD"
      >
        USD
      </option>
      <option
        selected=""
        value="CNY"
      >
        CNY
      </option>
      <option
        value="KRW"
      >
        KRW
      </option>
    </select>
  </div>
</div>
`;
